frontend/src/components/ExerciseModal/ExerciseModal.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState } from "react";
+import styles from "./ExerciseModal.module.css";
+
+interface ExerciseModalProps {
+  exercise: string;
+  onClose: () => void;
+}
+
+interface exerciseData {
+  name: string;
+  category: string;
+  equipment: string;
+  instructions: string[];
+  primaryMuscles: string[];
+  secondaryMuscles: string[];
+  images: string[];
+}
+
+const ExerciseModal: React.FC<ExerciseModalProps> = ({ exercise, onClose }) => {
+  const [exerciseData, setExerciseData] = useState<exerciseData>({
+    name: "",
+    category: "",
+    equipment: "",
+    instructions: [],
+    primaryMuscles: [],
+    secondaryMuscles: [],
+    images: [],
+  });
+
+  useEffect(() => {
+    const handleFetch = async (exerciseName: string) => {
+      try {
+        const options = {
+          method: "GET",
+          headers: {
+            "x-rapidapi-key": import.meta.env.VITE_RAPIDAPI_KEY,
+            "x-rapidapi-host": "exercise-db-fitness-workout-gym.p.rapidapi.com",
+          },
+        };
+        const response = await fetch(
+          `https://exercise-db-fitness-workout-gym.p.rapidapi.com/exercise/${exerciseName}`,
+          options
+        );
+
+        if (!response.ok) {
+          throw new Error("Failed to fetch exercise info");
+        }
+
+        const exerciseData = await response.json();
+        setExerciseData(exerciseData);
+      } catch (error) {
+        console.error("Error fetching exercise info:", error);
+      }
+    };
+    handleFetch(exercise);
+  }, []);
+
+  return (
+    <div className={styles.modalOverlay} onClick={onClose}>
+      <div className={styles.modalContent} onClick={(e) => e.stopPropagation()}>
+        <h2 className={styles.modalTitle}>{exerciseData.name}</h2>
+        <div className={styles.modalBody}>
+          <div className={styles.exerciseInfo}>
+            <p>
+              <strong>Category:</strong> {exerciseData.category}
```

Comment:
Should we just return null or have a loading indicator when exerciseData is null? When it's fetching the data, there is a quick status when the modal only shows empty data for all the info, and it looks a little confusing.

Reply:
Thanks for pointing that out. One of my project requirements is to add a loading indicator for visual polish. I am yet to work on loading screens so I will fix this by replacing it with a loader in a future PR